apiserver/facades/controller/charmrevisionupdater/interface_test.go

```diff
@@ -0,0 +1,210 @@
+	app.EXPECT().Channel().Return(csparams.Channel("latest/stable")).AnyTimes()
+	app.EXPECT().ApplicationTag().Return(names.ApplicationTag{Name: appID}).AnyTimes()
+
+	return app
+}
+
+func makeModel(c *gc.C, ctrl *gomock.Controller) charmrevisionupdater.Model {
+	model := mocks.NewMockModel(ctrl)
+	model.EXPECT().CloudName().Return("testcloud").AnyTimes()
+	model.EXPECT().CloudRegion().Return("juju-land").AnyTimes()
+	uuid := testing.ModelTag.Id()
+	cfg, err := config.New(true, map[string]interface{}{
+		"charm-hub-url": "https://api.staging.snapcraft.io", // not actually used in tests
+		"name":          "model",
+		"type":          "type",
+		"uuid":          uuid,
+	})
+	c.Assert(err, jc.ErrorIsNil)
+	model.EXPECT().Config().Return(cfg, nil).AnyTimes()
+	model.EXPECT().IsControllerModel().Return(false).AnyTimes()
+	model.EXPECT().UUID().Return(uuid).AnyTimes()
+	return model
+}
+
+func makeState(c *gc.C, ctrl *gomock.Controller, resources state.Resources) *mocks.MockState {
+	if resources == nil {
+		r := statemocks.NewMockResources(ctrl)
+		r.EXPECT().SetCharmStoreResources(gomock.Any(), gomock.Len(0), gomock.Any()).Return(nil).AnyTimes()
+		resources = r
+	}
+	state := mocks.NewMockState(ctrl)
+	state.EXPECT().Cloud(gomock.Any()).Return(cloud.Cloud{Type: "cloud"}, nil).AnyTimes()
+	state.EXPECT().ControllerUUID().Return("controller-1").AnyTimes()
+	state.EXPECT().Model().Return(makeModel(c, ctrl), nil).AnyTimes()
+	state.EXPECT().Resources().Return(resources, nil).AnyTimes()
+	return state
+}
+
+func makeResource(c *gc.C, name string, revision, size int, hexFingerprint string) resource.Resource {
+	fingerprint, err := resource.ParseFingerprint(hexFingerprint)
+	c.Assert(err, jc.ErrorIsNil)
+	return resource.Resource{
+		Meta: resource.Meta{
+			Name: name,
+			Type: resource.TypeFile,
+		},
+		Origin:      resource.OriginStore,
+		Revision:    revision,
+		Fingerprint: fingerprint,
+		Size:        int64(size),
+	}
+}
+
+// charmhubConfigMatcher matches only the charm IDs and revisions of a
+// charmhub.RefreshMany config.
+type charmhubConfigMatcher struct {
+	expected []charmhubConfigExpected
+}
+
+type charmhubConfigExpected struct {
+	id       string
+	revision int
+}
+
+func (m charmhubConfigMatcher) Matches(x interface{}) bool {
+	config, ok := x.(charmhub.RefreshConfig)
+	if !ok {
+		return false
+	}
+	request, _, err := config.Build()
+	if err != nil {
+		return false
+	}
+	if len(request.Context) != len(m.expected) || len(request.Actions) != len(m.expected) {
+		return false
+	}
+	for i, context := range request.Context {
+		if context.ID != m.expected[i].id {
+			return false
+		}
+		if context.Revision != m.expected[i].revision {
+			return false
+		}
+		action := request.Actions[i]
+		if *action.ID != m.expected[i].id {
+			return false
+		}
+	}
+	return true
+}
+
+func (charmhubConfigMatcher) String() string {
+	return "matches config"
+}
+
+func newFakeCharmstoreClient(st charmrevisionupdater.State) (charmstore.Client, error) {
```

Comment:
I'm not particularly bullish on using GoMock for everything - a test suite might only need one or two trivial fakes; but I think consistency within a suite is worth adopting. 

This one would be trivial as a generated mock.